FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc

```diff
@@ -16,18 +16,124 @@ Description: [one line class summary]
 //
 //
 
-// system include files
-#include <memory>
-
-// user include files
-#include "FastSimulation/CTPPSFastTrackingProducer/interface/CTPPSFastTrackingProducer.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Math/interface/Vector.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "FastSimDataFormats/CTPPSFastSim/interface/CTPPSFastRecHit.h"
+#include "FastSimDataFormats/CTPPSFastSim/interface/CTPPSFastRecHitContainer.h"
+#include "FastSimDataFormats/CTPPSFastSim/interface/CTPPSFastTrack.h"
+#include "FastSimDataFormats/CTPPSFastSim/interface/CTPPSFastTrackContainer.h"
+#include "FastSimulation/CTPPSFastGeometry/interface/CTPPSToFDetector.h"
+#include "FastSimulation/CTPPSFastGeometry/interface/CTPPSTrkDetector.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "Utilities/PPS/interface/PPSUnitConversion.h"
 #include "Utilities/PPS/interface/PPSUtilities.h"
+
 #include "TLorentzVector.h"
+
 // hector includes
 #include "H_Parameters.h"
 #include "H_BeamLine.h"
 #include "H_RecRPObject.h"
 #include "H_BeamParticle.h"
+
+class CTPPSFastTrackingProducer : public edm::stream::EDProducer<> {
+public:
+  explicit CTPPSFastTrackingProducer(const edm::ParameterSet&);
+  ~CTPPSFastTrackingProducer() override;
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
```

Comment:
This probably goes beyond this PR, but `beginStream()` and `endStream()` are empty and thus could be removed.

Reply:
Sure!